### ISSUE TITLE
[TxSubmitter] persistence

### ISSUE THREAD

**Issue by dappelt** [interledgerjs/ilp-plugin-xrp-paychan-shared#14]
The concern of transaction persistence was brought up [here](https://github.com/interledgerjs/ilp-plugin-xrp-paychan-shared/pull/10#issuecomment-376339798) and [here](https://github.com/interledgerjs/ilp-plugin-xrp-paychan-shared/pull/10#issuecomment-376381816).

I see two options for the persistence strategy:
1) Write the details of every transaction to disk before submission. In case of an unexpected shutdown, read back the transaction details from disk. Verify if the transaction succeeded and resubmit the transaction if necessary.

2) Once back-up, call [`getTransactions()`](https://github.com/ripple/ripple-lib/blob/develop/docs/index.md#gettransactions) to learn what happened to recently submitted transactions and initialize the application state accordingly.

@sharafian, what do you think?

**Comment by dappelt** [comment]
I thought about it some more and decided to implement 1)

**Comment by sharafian** [comment]
I agree that 1 sounds good. That way we have extra protection in case the transaction doesn't make it into the ledger at all
